git.go

```diff
@@ -0,0 +1,244 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/retry"
+)
+
+func isOriginPresent(dir, repoURL string) (bool, error) {
+	absDir, err := pathutil.AbsPath(dir)
+	if err != nil {
+		return false, err
+	}
+
+	if file, err := os.Stat(absDir); os.IsNotExist(err) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	} else if !file.IsDir() {
+		return false, fmt.Errorf("file (%s) exists, but it's not a directory", dir)
+	}
+
+	remotes, err := runForOutput(Git.RemoteList())
+	if err != nil {
+		return false, err
+	}
+
+	if !strings.Contains(remotes, repoURL) {
+		return false, fmt.Errorf(".git folder exists in the directory (%s), but using a different remote", dir)
+	}
+
+	return true, nil
+}
+
+func resetRepo() error {
+	if err := run(Git.Reset("--hard", "HEAD")); err != nil {
+		return err
+	}
+
+	if err := run(Git.Clean("-x", "-d", "-f")); err != nil {
+		return err
+	}
+
+	if err := run(Git.SubmoduleForeach(Git.Reset("--hard", "HEAD"))); err != nil {
+		return err
+	}
+
+	return run(Git.SubmoduleForeach(Git.Clean("-x", "-d", "-f")))
+}
+
+func isPR() bool {
+	return config.PRRepositoryCloneURL != "" || config.PRID != "" || config.PRMergeBranch != ""
+}
+
+func getCheckoutArg() string {
+	arg := ""
+	if config.Commit != "" {
+		arg = config.Commit
+	} else if config.Tag != "" {
+		arg = config.Tag
+	} else if config.Branch != "" {
+		arg = config.Branch
+	}
+	return arg
+}
+
+func getDiffFile() (string, error) {
+	url := fmt.Sprintf("%s/diff.txt?api_token=%s", config.BuildURL, config.BuildAPIToken)
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	} else if resp.StatusCode != 200 {
+		return "", fmt.Errorf("Can't download diff file, HTTP status code: %d", resp.StatusCode)
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
```

Comment:
this should be right after `if err != nil` -> you have to close the body in every case if there's no `err`, not just when you read from the body